server/events/plan_command_runner.go

```diff
@@ -168,6 +171,15 @@ func (p *PlanCommandRunner) run(ctx *command.Context, cmd *CommentCommand) {
 	baseRepo := ctx.Pull.BaseRepo
 	pull := ctx.Pull
 
+	ctx.PullRequestStatus, err = p.pullReqStatusFetcher.FetchPullStatus(pull)
```

Comment:
Why is this needed?

Reply:
This is addressing @krrrr38's comment. For this change we are checking approved and mergeability states during plan and those checks require ctx.PullRequestStatus to be populated. What I saw when testing this without these lines is that the plan was correctly blocked by the approval check, but it would never succeed as the approval status was not updated during a plan run.

Reply:
This field is useful and we forget to fill-in. So `ctx.PullRequestStatus` might be better to fill-in at the instance creation timing https://github.com/runatlantis/atlantis/blob/cb0aadf57127d67223483b7ab2184df8b2671a5f/server/events/command_runner.go#L150-L158

But it requires apply_command_runer/import_command_runner change too, so I think it doesn't need it in this PR.

Reply:
can this comment resolved?

Reply:
The above impl is optional in this PR and not need to do it in this PR, so resolved 👍